Editor/Data/Type/DefaultObservableListDrawer.cs

```diff
@@ -2,139 +2,30 @@
-                        {
-                            dynamic currentElement = elements[index];
-                            Type elementType = currentElement == null ? GetElementType() : currentElement.GetType();
-                            using (new EditorGUILayout.HorizontalScope())
-                            {
-                                EditorGUILayout.PrefixLabel(string.Format(collectionElementLabel, index));
-                                using (new EditorGUI.IndentLevelScope(indentResetLevel))
-                                using (EditorGUI.ChangeCheckScope check = new EditorGUI.ChangeCheckScope())
-                                {
-                                    dynamic elementValue = EditorGUILayout.ObjectField(currentElement, elementType, true);
-                                    if (check.changed)
-                                    {
-                                        list.SetAt(elementValue, index);
-                                    }
-                                }
-                            }
-                        }
-                    }
-
-                    using (new EditorGUILayout.HorizontalScope())
-                    {
-                        GUILayout.FlexibleSpace();
-                        if (GUILayout.Button(buttonIconAdd, buttonStyle))
-                        {
-                            list.Add(null);
-                        }
-
-                        using (new EditorGUI.DisabledScope(isListEmpty))
-                        {
-                            if (GUILayout.Button(buttonIconRemove, buttonStyle))
-                            {
-                                list.RemoveAt(list.NonSubscribableElements.Count - 1);
-                            }
-                        }
-
-                        GUILayout.Space(objectPickerIcon.image.width - objectPickerPadding);
-                    }
-
-                    GUILayout.Space(footerSpacing);
-                }
-            }
-        }
-
-        public override float GetPropertyHeight(SerializedProperty property, GUIContent label)
-        {
-            return base.GetPropertyHeight(property, label) - EditorGUIUtility.singleLineHeight;
-        }
-
-        protected virtual bool IsSupportedElement(dynamic element)
-        {
-            return element == null || typeof(UnityEngine.Object).IsAssignableFrom(element.GetType());
-        }
-
-        protected virtual Type GetElementType()
-        {
-            Type type = fieldInfo.FieldType;
-            while (type.BaseType != null)
-            {
-                type = type.BaseType;
-                if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(ObservableList<,>))
-                {
-                    return type.GenericTypeArguments[0];
-                }
+                SerializedObject referenceObject = new SerializedObject(propertyObject as Object);
```

Comment:
Use a direct cast instead. Or, if the above suggestion works, there is no need for a cast here in the first place.